mods/gen1/statuses.js

```diff
@@ -310,8 +310,8 @@ exports.BattleStatuses = {
 				target.removeVolatile('Protect');
 				target.removeVolatile('Endure');
 
-				if (typeof posData.moveData.affectedByImmunities === 'undefined') {
-					posData.moveData.affectedByImmunities = true;
+				if (typeof posData.moveData.ignoreImmunity === 'undefined') {
```

Comment:
`typeof ... === 'undefined'` should probably be rewritten `... === undefined` or perhaps `!(... in ...)`.

I wrote that that way back when I didn't really understand JavaScript.